plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1333,6 +1335,78 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
         }
     }
 
+    @Override
+    public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        DeltaLakeTableHandle table = (DeltaLakeTableHandle) tableHandle;
+        DeltaLakeColumnHandle deltaLakeColumn = (DeltaLakeColumnHandle) columnHandle;
+        String dropColumnName = deltaLakeColumn.getBaseColumnName();
+        MetadataEntry metadataEntry = table.getMetadataEntry();
+
+        checkSupportedWriterVersion(session, table);
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(metadataEntry);
+        if (columnMappingMode != ColumnMappingMode.NAME && columnMappingMode != ColumnMappingMode.ID) {
+            throw new TrinoException(NOT_SUPPORTED, "Cannot drop column with the column mapping: " + columnMappingMode);
+        }
+
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, table);
+        long commitVersion = table.getReadVersion() + 1;
+        List<String> partitionColumns = getPartitionedBy(tableMetadata.getProperties());
+        if (partitionColumns.contains(dropColumnName)) {
+            throw new TrinoException(NOT_SUPPORTED, "Cannot drop partition column: " + dropColumnName);
+        }
+
+        List<DeltaLakeColumnMetadata> columns = extractSchema(metadataEntry, typeManager);
+        ImmutableList.Builder<String> columnNames = ImmutableList.builderWithExpectedSize(columns.size() - 1);
+        ImmutableMap.Builder<String, String> physicalColumnNameMappingBuilder = ImmutableMap.builderWithExpectedSize(columns.size());
+        for (DeltaLakeColumnMetadata column : columns) {
+            DeltaLakeColumnHandle handle = toColumnHandle(column.getColumnMetadata(), column.getName(), column.getType(), partitionColumns);
+            if (!column.getName().equals(dropColumnName)) {
+                columnNames.add(handle.getColumnName());
+            }
+            physicalColumnNameMappingBuilder.put(column.getName(), column.getPhysicalName());
+        }
+        Map<String, String> physicalColumnNameMapping = physicalColumnNameMappingBuilder.buildOrThrow();
+
+        Map<String, Object> columnTypes = filterKeys(getColumnTypes(metadataEntry), name -> !name.equals(dropColumnName));
+        Map<String, String> columnComments = filterKeys(getColumnComments(metadataEntry), name -> !name.equals(dropColumnName));
```

Comment:
are map keys lowercase or original?

Reply:
Original. Changed to `equalsIgnoreCase` and added `testUnsupportedDropColumnWithColumnMappingModeNone`.